Tests/SwinjectAutoregistrationTests/AutoregistrationTests.swift

```diff
@@ -0,0 +1,281 @@
+    }
+    
+    func testServiceIsRegisteredWithEightDependencies() {
+        container.autoregister(Service8.self, initializer: Service8.init)
+        let service = container.resolve(Service8.self)
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithNineDependencies() {
+        container.autoregister(Service9.self, initializer: Service9.init)
+        let service = container.resolve(Service9.self)
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithTwentyDependencies() {
+        container.autoregister(Service20.self, initializer: Service20.init)
+        let service = container.resolve(Service20.self)
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithOneDynamicArgument() {
+        container.autoregister(Service2.self, argument: DependencyB.self, initializer: Service2.init)
+        let service = container.resolve(Service2.self, argument: DependencyB())
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithTwoDynamicArguments() {
+        container.autoregister(Service3.self, arguments: DependencyB.self, DependencyC.self, initializer: Service3.init)
+        let service = container.resolve(Service3.self, arguments: DependencyB(), DependencyC())
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithThreeDynamicArguments() {
+        container.autoregister(Service4.self, arguments: DependencyB.self, DependencyC.self, DependencyD.self, initializer: Service4.init)
+        let service = container.resolve(Service4.self, arguments: DependencyB(), DependencyC(), DependencyD())
+        XCTAssertNotNil(service)
+    }
+    
+    func testServiceIsRegisteredWithFourDynamicArguments() {
+        container.autoregister(Service4.self, arguments: DependencyC.self, DependencyD.self, DependencyB.self, initializer: Service4.init)
+        let service = container.resolve(Service4.self, arguments: DependencyC(), DependencyD(), DependencyB())
+        XCTAssertNotNil(service)
+    }
+
+    func testServiceIsRegisteredWithOptionalDependencies() {
+        container.autoregister(OptionalService.self, initializer: OptionalService.init)
+        let service = container.resolve(OptionalService.self)
+        XCTAssertNotNil(service)
+    }
+
+    func testServiceIsRegisteredWithUnwrrappedDependencies() {
+        container.autoregister(UnwrappedService.self, initializer: UnwrappedService.init)
+        let service = container.resolve(UnwrappedService.self)
+        XCTAssertNotNil(service)
+    }
+
+    #if !SWIFT_PACKAGE && arch(x86_64)
+    func testAssertionIsThrownWhenSameTypeArgumentsArePassed() {
+        // For now we just comment out the test for `precondition()` because we cannot do it without depending on Nimble or CwlPreconditionTesting.
+        
+        // container.autoregister(SameArgumentsService.self, arguments: String.self, Int.self, String.self, initializer: SameArgumentsService.init)
```

Comment:
Commented out `precondition` test here.

Reply:
Should we think about maybe making this a `throw`ing function that throws a custom error when the arguments are not unique types? That would make it easier to test. We could also mock `precondition()` using an injected closure, but I think making this `throw`ing is the best way.

Reply:
`throw`ing an error is a solution if we accept a breaking change (The users of the API must add `try` where `autoregister` is used). If we take the approach, SwinjectAutoregistration version will go to v3.

Another approach is adding a dependency on `CwlPreconditionTesting`. The project is pretty stable where the last release was on Oct 24, 2019 but it keeps working used by the latest versions of Quick/Nimble. The dependency on `CwlPreconditionTesting` looks not so bad.